components/board/customlinux/board.go

```diff
@@ -87,13 +87,13 @@ func parseRawPinData(pinData []byte, filePath string) ([]genericlinux.PinDefinit
 	for name, pin := range parsedPinData.Pins {
 		err = multierr.Combine(err, pin.Validate(filePath))
 
-		// Until we get hardware PWM working reliably on lots of boards (likely by being able to
-		// set the pin mode directly), we disable all hardware PWM on these boards.
-		// The range operator makes a copy of the value, so to change the original, we need to look
-		// it up in the map again.
-		// TODO(RSDK-5105): Undo this when we're ready to support hardware PWM.
-		parsedPinData.Pins[name].PwmChipSysfsDir = ""
-		parsedPinData.Pins[name].PwmID = -1
+		// Until we can reliably switch between gpio and pwm on lots of boards, pins that have
+		// hardware pwm enabled will be hardware pwm only. Disabling gpio functianality on these
+		// pins.
+		if parsedPinData.Pins[name].PwmChipSysfsDir != "" {
+			logging.Global().Warnf("pin %s can be used for PWM only", parsedPinData.Pins[name].Name)
```

Comment:
can we add a to-do ticket to refactor the generic Linux constructor so it looks more like other component constructors? That way we can pass in the rdk logger and not have to rely on a global logger.